Userland/Libraries/LibELF/DynamicLinker.cpp

```diff
@@ -108,7 +110,22 @@ static Result<NonnullRefPtr<DynamicLoader>, DlErrorMessage> map_library(const St
         return map_library(name, fd);
     }
 
-    // TODO: Do we want to also look for libs in other paths too?
+    // Scan the LD_LIBRARY_PATH envvar if it exists or we're not a SUID program
+    if (s_ld_library_path != nullptr) {
+        auto paths = StringView { s_ld_library_path }.split_view(':');
+        for (const auto& search_path : paths) {
+            StringBuilder path_builder;
+            path_builder.append(search_path);
+            path_builder.append(name);
```

Comment:
Hmm, LD_LIBRARY_PATH=/some/dir should also work, but doesn't in this case, it might be worth it to use LexicalPath here?

Reply:
Yes, please use LexicalPath for any path transformation instead of raw string shenanigans :P